packages/react/src/useActionForm.ts

```diff
@@ -0,0 +1,293 @@
+  const [findResult] = useFindExistingRecord(modelManager, options?.findBy || "1", {
+    pause: !findExistingRecord,
+    select: options?.select,
+  });
+
+  const isReady = !findExistingRecord || !!findResult.data;
+
+  let defaultValues = options?.defaultValues;
+  if (!defaultValues && "modelApiIdentifier" in action && findResult.data) {
+    defaultValues = { [action.modelApiIdentifier]: { ...toDefaultValues(action.modelApiIdentifier, findResult.data) } } as any;
+  }
+
+  let existingRecordId: string | undefined = undefined;
+
+  if (findResult.data?.id) {
+    existingRecordId = findResult.data.id;
+  } else if (defaultValues?.id) {
+    existingRecordId = defaultValues.id;
+  } else if ("modelApiIdentifier" in action && defaultValues?.[action.modelApiIdentifier]?.id) {
+    existingRecordId = defaultValues?.[action.modelApiIdentifier]?.id;
+  }
+
+  // setup the react-hook-form object, passing any options from the props
+  const { handleSubmit, formState, ...formHook } = useForm<ActionFunc["variablesType"], FormContext>({
+    ...options,
+    defaultValues: toDefaultValues("modelApiIdentifier" in action ? action.modelApiIdentifier : undefined, defaultValues),
+  });
+
+  // when the default values arrive from the record find later, reset them into the form. react-hook-form doesn't watch the default values after the first render
+  useEffect(() => {
+    if (isReady && !hasSetInitialValues.current && defaultValues) {
+      hasSetInitialValues.current = true;
+      formHook.reset(defaultValues);
+    }
+  }, [isReady, defaultValues, formHook]);
+
+  // get the action runner to run on submit
+  const [actionResult, runAction] =
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    "modelApiIdentifier" in action ? useAction(action, { select: options?.select } as any) : useGlobalAction(action);
+
+  const handleSubmissionError = useCallback(
+    (error: Error | FieldErrors<ActionFunc["variablesType"]>) => {
+      if ("executionErrors" in error) {
+        const executionErrors = (error as unknown as ErrorWrapper).executionErrors;
+        for (const executionError of executionErrors) {
+          if ("validationErrors" in executionError) {
+            for (const validationError of executionError.validationErrors) {
+              const errorKey =
+                "modelApiIdentifier" in action
+                  ? `${action.modelApiIdentifier}.${validationError.apiIdentifier}`
+                  : validationError.apiIdentifier;
+              formHook.setError(errorKey as any, {
+                message: validationError.message,
+              });
+            }
+          } else {
+            const codeToReplace = "code" in executionError ? `${executionError.code}: ` : "";
+            const message = executionError.message.replace(codeToReplace, "");
+
+            formHook.setError("root", { message });
```

Comment:
I like this pattern, didn't notice you added this! I think we should put transport errors in here too if we're gonna do it like this, like if the network is disconnected or what have you. Can you verify that's working ok?

Reply:
yah this was a big win for the auth templates; transport here meaning no `code`?

Reply:
Yes no `code`, but also I think the if statement at the top here concerning `"executionErrors" in error` might be ignoring errors where the network doesn't work at all. I think that "executionErrors" is an `urql` thing, and there can also be a `networkError` that comes when there's no connection made at all or a timeout or whatever. I think the move would just be to add a test and see what happens. I took a look but I couldn't find a case where we do this kind of transport error already :(

Reply:
yah was looking for that; I think `executionErrors` is our thing?

```/**
   * A list of errors encountered by the backend when processing a request. Populated if the client successfully communicated with the backend, but the backend was unable to process the request and rejected it with an error.
   * Includes GraphQL syntax errors, missing or invalid argument errors, data validation errors, or unexpected errors encountered when running backend logic.
   **/
```

I think maybe we don't want the network errors in the `formState`? I think this would be the reason to have the top level `error` as well... i.e. you watch that to put up a something went wrong toast, but you don't show the network error as a form server side validation error

Reply:
I found a `mockNetworkError`

Reply:
ok added a test asserting that 

```
expect((result.current.error as any).message).toMatchInlineSnapshot(`"[Network] Network error"`);
expect(result.current.formState.errors).toEqual({});
```

I think that's right to me, showing network error as a form error seems wrong, but I don't feel strongly about that

Reply:
ok added the network errors to the root